cvat-core/src/server-proxy.ts

```diff
@@ -1065,48 +1067,89 @@ async function restoreProject(storage: Storage, file: File | string) {
     return wait();
 }
 
-async function createTask(taskSpec, taskDataSpec, onUpdate) {
-    const { backendAPI, origin } = config;
-    // keep current default params to 'freeze" them during this request
-    const params = enableOrganization();
+const listenToCreateCallbacks: Record<number, {
+    promise: Promise<SerializedTask>;
+    onUpdate: ((state: string, progress: number, message: string) => void)[];
+}> = {};
 
-    async function wait(id) {
-        return new Promise((resolve, reject) => {
-            async function checkStatus() {
-                try {
-                    const response = await Axios.get(`${backendAPI}/tasks/${id}/status`, { params });
-                    if (['Queued', 'Started'].includes(response.data.state)) {
-                        if (response.data.message !== '') {
-                            onUpdate(response.data.message, response.data.progress || 0);
-                        }
-                        setTimeout(checkStatus, 1000);
-                    } else if (response.data.state === 'Finished') {
-                        resolve();
-                    } else if (response.data.state === 'Failed') {
-                        // If request has been successful, but task hasn't been created
-                        // Then passed data is wrong and we can pass code 400
-                        const message = `
-                            Could not create the task on the server. ${response.data.message}.
-                        `;
-                        reject(new ServerError(message, 400));
-                    } else {
-                        // If server has another status, it is unexpected
-                        // Therefore it is server error and we can pass code 500
-                        reject(
-                            new ServerError(
-                                `Unknown task state has been received: ${response.data.state}`,
-                                500,
-                            ),
+function listenToCreateTask(
+    id, onUpdate: (state: RQStatus, progress: number, message: string) => void,
+): Promise<SerializedTask> {
+    if (id in listenToCreateCallbacks) {
+        listenToCreateCallbacks[id].onUpdate.push(onUpdate);
+        // to avoid extra status check requests we do not create any more promises
+        return listenToCreateCallbacks[id].promise;
+    }
+
+    const promise = new Promise<SerializedTask>((resolve, reject) => {
+        const { backendAPI } = config;
+        const params = enableOrganization();
+        async function checkStatus(): Promise<void> {
+            try {
+                const response = await Axios.get(`${backendAPI}/tasks/${id}/status`, { params });
+                if (['Queued', 'Started'].includes(response.data.state)) {
```

Comment:
Why dont we use `RQStatus` enum here? Should server return status in lowercase?

Reply:
> Why dont we use RQStatus enum here? 

Exactly because server returns statuses not in lower case. 
I do not want to change API scheme

Reply:
But I suppose the statuses returned from server are the ones in RQStatus, we can cast it to lowercase on UI

Reply:
Done